packages/grid/src/styled/splitter/StyledSeparatorContainer.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { retrieveComponentStyles, DEFAULT_THEME, getColor } from '@zendeskgarden/react-theming';
+import styled, { css, ThemeProps, DefaultTheme } from 'styled-components';
+import { math } from 'polished';
+import { StyledSeparator } from './StyledSeparator';
+
+const COMPONENT_ID = 'splitter.separator_container';
+
+interface IStyledSeparatorContainerProps {
+  isHorizontal?: boolean;
+}
+
+const sizeStyles = ({
+  isHorizontal,
+  theme
+}: IStyledSeparatorContainerProps & ThemeProps<DefaultTheme>) => {
+  const containerMarginCorrection = math(`${theme.shadowWidths.md} + ${theme.borderWidths.sm}`);
+  const containerThickness = math(`${theme.shadowWidths.md} * 2 + ${theme.borderWidths.sm}`);
+
+  let containerWidth;
+  let containerHeight;
+  let marginDirection;
+  let flexPositioning;
+  let cursorType;
+  let separatorHeight;
+  let separatorWidth;
+
+  if (isHorizontal) {
+    marginDirection = 'margin-top';
+    containerWidth = '100%';
+    containerHeight = containerThickness;
+    separatorHeight = theme.shadowWidths.md;
+    flexPositioning = 'align-items';
+    cursorType = 'row-resize';
+  } else if (theme.rtl) {
+    marginDirection = 'margin-right';
+    containerWidth = containerThickness;
+    separatorWidth = theme.shadowWidths.md;
+    containerHeight = '100%';
+    flexPositioning = 'justify-content';
+    cursorType = 'col-resize';
+  } else {
+    marginDirection = 'margin-left';
+    containerWidth = containerThickness;
+    separatorWidth = theme.shadowWidths.md;
+    containerHeight = '100%';
+    flexPositioning = 'justify-content';
+    cursorType = 'col-resize';
+  }
+
+  return css`
+    cursor: ${cursorType};
+    width: ${containerWidth};
+    height: ${containerHeight};
+    ${flexPositioning}: center;
+    ${marginDirection}: -${containerMarginCorrection};
+
+    &:hover > ${StyledSeparator} {
+      width: ${separatorWidth};
+      height: ${separatorHeight};
+    }
+
+    &:focus {
+      outline: none;
+    }
+
+    &:focus > ${StyledSeparator} {
```

Comment:
Nearly all focus styling in Garden is handled via `&[data-garden-focus-visible]` rather than `&:focus`. Compare with https://github.com/zendeskgarden/react-components/blob/9fb47d9118bcc82d35169a36c446241a98b4beb8/packages/tabs/src/styled/StyledTab.ts#L38-L40 We get some nice a11y styling goodness by relying on `ThemeProvider`'s [:focus-visible](https://garden.zendesk.com/components/theme-provider#themeprovider) polyfill.

Reply:
Fixed in your changes